### DISCUSSION TITLE
Best Way To Handle Checking For Duplicate Registrations

### DISCUSSION THREAD

**Discussion by anthony-mg** [pocketbase/pocketbase#3719] [Q&A] [answered]
I have an app that saves a Spotify access token to the DB when the user registered, and also gets a user_id from spotify. I have no passwords or usernames/email auths, I just store the access token and refresh token. I want to check if the user is already registered before adding a new access token to the table "Tokens". Currently I am doing:
```
try{
   const record = $app
      .dao()
      .findFirstRecordByData("Tokens", "user_id", "non_existant_user_id");
} catch(error){
   if (error.message === "sql: no rows in result set") {
      //add the user
    }
}
```
This will jump to the catch block and add the new user. This smells very anti-pattern as I am using errors as control flow. Is there a better way of doing this? In head the way it would work would be if no row is found, record would be an empty object, but it does not seem to work like that. Maybe I am missing crucial sqlite experience?

**Comment by ganigeorgiev** [comment, marked as answer]
If all you want is a unique field validation then you can create a unique index for your field.

This can be done from the _Admin UI > Collections > Edit collection_ panel under the "Unique constraints and indexes" section:
![indexes_screenshot](https://camo.githubusercontent.com/068c2f4d7fa52945b69fbbbcb887b521171f46a1fe387630519ee51fa80ae9c0/68747470733a2f2f692e696d6775722e636f6d2f6a617369434c4d2e706e67)

**Reply by ganigeorgiev** [reply]
I'm confused.

> Though, I do want to notify the client that this is a duplicate

The unique index should return API validation error for the field under `err.response.data.yourField`.

> since saveRecord() returns void...

If you have created the unique index then both via the web APIs and calling manually `saveRecord(record)` should throw an error if you are trying to create a new record with duplicated field value.

_Edit: it seems that you've deleted your comment after I posted mine._

**Reply by anthony-mg** [reply]
Sorry I thought I had been quick enough when deleting my reply, I realized afterwards saveRecord does in fact throw an error, makes the control-flow a bit more sensible. Thanks for the help!
